src/contract_wrappers/exchange_wrapper.ts

```diff
@@ -147,28 +147,41 @@ export class ExchangeWrapper extends ContractWrapper {
     public async fillOrderAsync(signedOrder: SignedOrder, fillTakerTokenAmount: BigNumber.BigNumber,
                                 shouldThrowOnInsufficientBalanceOrAllowance: boolean,
                                 takerAddress: string): Promise<BigNumber.BigNumber> {
+        // validate input types
         assert.doesConformToSchema('signedOrder', signedOrder, schemas.signedOrderSchema);
         assert.isBigNumber('fillTakerTokenAmount', fillTakerTokenAmount);
         assert.isBoolean('shouldThrowOnInsufficientBalanceOrAllowance', shouldThrowOnInsufficientBalanceOrAllowance);
-        await assert.isSenderAddressAsync('takerAddress', takerAddress, this._web3Wrapper);
```

Comment:
Why was this assertion removed?

Reply:
It was my mistake to include this change in this PR, as it contributes to overall optimization of this method but does not relate to parallelizng RPCs, I still believe it should be deduplicated with the isSenderAddressAsync call in validateFillOrderThrowIfInvalidAsync. We should try to save as many RPCs as possible especially if the web3Provider is working with a remote ethereum client

Reply:
Hm... My issue with that is `validateFillOrderThrowIfInvalidAsync` is called internally within `fillOrderAsync`. As part of the implementation, it is subject to change and could be removed in a subsequent refactor. I feel strongly that all public interface assertions should exist in the public method itself. I think a more appropriate refactor would be to call `this._orderValidationUtils.validateFillOrderThrowIfInvalidAsync` directly from this method rather then `validateFillOrderThrowIfInvalidAsync`.

Reply:
I think I figured it out, `isSenderAddressAsync` is doing the task of talking to an ethereum client and verifying that it exists,  that should be left in `validateFillOrderThrowIfInvalidAsync` as is. We should change this line to `assert.isETHAddressHex()` I believe that is more in line with what the rest of the assertions are doing, as well as saves us 1 RPC =)

As a side note I noticed that `isETHAddressHex` instantiates an instance of a web3 object, is this a free operation? we may be better off passing in an existing instance of web3 or wrapper, similar to the interface of  `isSenderAddressAsync`